components/LocalBox.php

```diff
@@ -166,7 +166,7 @@ public function onChangeOrderType()
             if ($cartBox AND $cartBox->property('pageIsCheckout'))
                 return Redirect::to($this->controller->pageUrl($this->property('checkoutPage')));
 
-            return array_merge($cartBox->fetchPartials(), $this->fetchPartials());
+            return Redirect::to($this->controller->pageUrl($this->property('menusPage')));
```

Comment:
You can use Redirect:back() here to redirect to the current page, so users changing order type on mobile or during checkout don't get redirected back to the menus page. The conditional above would no longer be needed making the `cartBoxAlias` property redundant.

Reply:
Cool - updated